src/main.js

```diff
@@ -34,14 +34,15 @@ var main = function() {
     return;
   }
 
+  Memory.myRooms = _(Game.rooms).filter(r => r.execute()).map(r => r.name).value();
```

Comment:
Without major rewrites in the core logic, there are essentially two options, with a couple of variants I can see:

1. if undefined => define as empty array and after doing action that depends on it, get all rooms
2. Get all rooms before you perform actions that depends on it.

2 has the drawback of a room that was just recently captured appearing with a potential one tick delay in the statistics, since you query in the beginning of the gametick instead of the end. It still is possible that ownership to rooms gets assigned by the server between the actual ticks though, this is something i do not have checked.

1 has the drawback of causing a potential performance decrease somewhere in the neighborhood of a nanosecond (or a fraction thereof).

Realistically speaking though, both options perform realistically the same.